### DISCUSSION TITLE
nameslist.txt

### DISCUSSION THREAD

**Discussion by LarsNauheimer** [chrisjackson-pellicle/HybPiper-RBGV#4] [General]
Is there the option to read in a file with samples that should be included in the assembly? 
I like the option in HybPiper running it with a nameslist file. as somethimes one might not want to have all files in the folder assembled.

**Comment by chrisjackson-pellicle** [comment]
Not at this stage. We could set it up to override the `namelist.txt` automatically produced by the pipeline if the user provides the file, and then Nextflow's `filter` operator (https://www.nextflow.io/docs/latest/operator.html#filtering-operators) looks like it'll be able to do the job. I'll add this to the to-do list...

**Reply by chrisjackson-pellicle** [reply]
@LarsNauheimer 

I'm gonna allege that this is working in the development branch script now (`git pull; git checkout remotes/origin/development`). Let me know if you get a chance to try it and if there are any issues. Note that I've only added this if providing either paired-end files or single-end only.

Provide your `namelist.txt` file using e.g. `--namelist namelist.txt`. The sample names in the file should correspond to the unique prefix of the read files used to match and/or group them. So, if you're providing the files:

     79678_LibID81729_HF7CKAFX2_TGAATGCC-TGTCTAGT_L001_R1.fastq
     79678_LibID81729_HF7CKAFX2_TGAATGCC-TGTCTAGT_L001_R2.fastq
     79679_LibID81730_HF7CKAFX2_GCAACTAT-TCGTTGAA_L001_R1.fastq
     79679_LibID81730_HF7CKAFX2_GCAACTAT-TCGTTGAA_L001_R2.fastq
     79680_LibID81730_HF7CKAFX2_GCAACTAT-TCGTTGAA_L001_R1.fastq
     79680_LibID81730_HF7CKAFX2_GCAACTAT-TCGTTGAA_L001_R2.fastq

...and you just want to process samples beginning `79678` and `79679`, your `namelist.txt` file should contain:

     79678_LibID81729_HF7CKAFX2_TGAATGCC-TGTCTAGT_L001
     79679_LibID81730_HF7CKAFX2_GCAACTAT-TCGTTGAA_L001

Note that if you're [combining reads files][1] that've been run across multiple lanes, your `namelist.txt` file should contain the prefix that will be used to group them (defaults to the first field of read filenames split by underscore). So, for read files:


     79678_LibID81729_HF7CKAFX2_TGAATGCC-TGTCTAGT_L001_R1.fastq
     79678_LibID81729_HF7CKAFX2_TGAATGCC-TGTCTAGT_L001_R2.fastq
     79678_LibID81729_HF7CKAFX2_TGAATGCC-TGTCTAGT_L002_R1.fastq
     79678_LibID81729_HF7CKAFX2_TGAATGCC-TGTCTAGT_L002_R2.fastq
     79679_LibID81730_HF7CKAFX2_GCAACTAT-TCGTTGAA_L001_R1.fastq
     79679_LibID81730_HF7CKAFX2_GCAACTAT-TCGTTGAA_L001_R2.fastq
     79679_LibID81730_HF7CKAFX2_GCAACTAT-TCGTTGAA_L002_R1.fastq
     79679_LibID81730_HF7CKAFX2_GCAACTAT-TCGTTGAA_L002_R2.fastq
     79680_LibID81731_HF7CKAFX2_GCAACTAT-TCGTTGAA_L001_R1.fastq
     79680_LibID81731_HF7CKAFX2_GCAACTAT-TCGTTGAA_L001_R2.fastq
     79680_LibID81731_HF7CKAFX2_GCAACTAT-TCGTTGAA_L002_R1.fastq
     79680_LibID81731_HF7CKAFX2_GCAACTAT-TCGTTGAA_L002_R2.fastq

...your `namelist.txt` should contain e.g.:

     79678
     79679
     
I look forward to the inevitable bugs...

[1]: https://github.com/chrisjackson-pellicle/HybPiper-RBGV/wiki/Additional-pipeline-features-and-details#combining-read-files-for-samples-run-across-multiple-lanes

**Reply by LarsNauheimer** [reply]
Hi Chris,
Thanks for implementing that! 
I only now got to try it and it worked smoothly so far. :)
